### ISSUE TITLE
Copyedit and organize decK best practices documentation

### ISSUE THREAD

**Issue by lena-larionova** [Kong/docs.konghq.com#3247]
### Where is the problem?

https://docs.konghq.com/deck/1.8.x/guides/best-practices/

### What happened?

The best practices page for decK is a jumbled list of information. It's hard to parse in its current state and could use some organization. At the same time, it needs a writer's touch to clean up the language for clarity and readability.

File to edit: https://github.com/Kong/docs.konghq.com/blob/main/app/deck/1.8.x/guides/best-practices.md

**Examples from the page**
* A run-on sentence that isn't using proper punctuation:
  "Declarative configuration is for … configuration, it is not meant for end-user data, which can easily grow into hundreds of thousands or millions."
* "Process" is singular when it should be plural: "Multiple process will step on each other"

### What did you expect to happen?

Clear, simple language in accordance with our style guide.

### Code of Conduct and Community Expectations

- [X] I agree to follow this project's Code of Conduct
- [X] I agree to abide by the Community Expectations

**Comment by Sayanta66** [comment]
Can I take this issue please? @lena-larionova Please assign it to me

**Comment by Rutam21** [comment]
@lena-larionova I was done with the 2 issues I was assigned to. So, I pretty much picked this up and pushed the required changes for this Issue as well. Feel free to assign this to me.

**Comment by lena-larionova** [comment]
Closing since this is complete now. Thanks for your contribution @Rutam21 ! Don't forget to submit your PR for consideration to the Kong Summit Hackathon: https://konghq.com/summit-hackathon/
